snap/naming/tag.go

```diff
@@ -0,0 +1,125 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package naming
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+var errInvalidSecurityTag = errors.New("invalid security tag")
+
+// ParsedSecurityTag describes a parsed snap security tag.
+type ParsedSecurityTag interface {
+	// String returns the entire security tag.
+	String() string
+
+	// InstanceName returns the snap name and instance key.
+	InstanceName() string
+}
+
+// ParsedAppSecurityTag describes a parsed snap application security tag.
+type ParsedAppSecurityTag interface {
```

Comment:
AppSecurityTag?

Reply:
I was wondering about naming and chose to repeat `Parsed` as a conservative choice. Since the usage of this interface is not widespread yet, for obvious reasons, I wanted to consider having a `type SecurityTag string` type that gets passed around and used were all we need is the entire string and parsing specific fields is not required, but having a distinct type might help for type safety and self-documentation. With that new hypothetical type, what is the relationship to the interface?

Reply:
+1 for AppSecurityTag / HookSecurityTag.